tests/test_samples.py

```diff
@@ -1339,3 +1339,140 @@ def test_old_gui():
         make_2d_axes(['x0', 'y0'], tex={'x0': '$x_0$', 'y0': '$y_0$'})
     with pytest.raises(NotImplementedError):
         make_1d_axes(['x0', 'y0'], tex={'x0': '$x_0$', 'y0': '$y_0$'})
+
+
+def test_groupby_stats():
+    mcmc = read_chains('./tests/example_data/cb')
+    chains = mcmc.groupby(('chain', '$n_\\mathrm{chain}$'))
+
+    assert chains.mean().isweighted() is True
+    assert chains.std().isweighted() is True
+    assert chains.median().isweighted() is True
+    assert chains.var().isweighted() is True
+    assert chains.kurt().isweighted() is True
+    assert chains.kurtosis().isweighted() is True
+    assert chains.skew().isweighted() is True
+    assert chains.mad().isweighted() is True
+    assert chains.sem().isweighted() is True
+    assert chains.corr().isweighted() is True
+    assert chains.cov().isweighted() is True
+    assert chains.corrwith(mcmc).isweighted() is True
```

Comment:
I've added these tests that ensure that our results from groupby methods are weighted. 

As you can see from https://github.com/williamjameshandley/anesthetic/pull/272/commits/142740cff80e44ca74590ccb6f474515dfe80583 the tests currently pass for `mean`, `std`, `median`, `var`, `kurt`, `kurtosis`, `skew`, and `sem`.

https://github.com/williamjameshandley/anesthetic/pull/272/commits/7b0a8e1aa233582d6d93679c378058154b04cba9 shows that `mad`, `corr`, `cov`, and `corrwith` are still failing, so I've unchecked them on our to-do list.

While the tests further down show that `cov` and Co. produce correct results within their respective group/chain, the tests don't show that the groups/chains are not correctly weighted with respect to one another.